zeppelin-web/src/app/interpreter/interpreter.controller.js

```diff
@@ -22,6 +22,171 @@ angular.module('zeppelinWebApp').controller('InterpreterCtrl',
     $scope.showRepositoryInfo = false;
     $scope._ = _;
 
+    // auto complete related
+    $scope.suggestions = [];
+    $scope.selectIndex = -1;
+    var selectedUser = '';
+    var selectedUserIndex = 0;
+    var previousSelectedList = [];
+    var searchText = [];
+
+    function updatePreviousList() {
+      for (var i = 0; i < searchText.length; i++) {
+        previousSelectedList[i] = searchText[i];
+      }
+    }
+
+    function convertToString(setting) {
+      setting.option.users = searchText.join();
+    }
+
+    var checkIfSelected = function() {
+      if (($scope.suggestions.length === 0) &&
+        ($scope.selectIndex < 0 || $scope.selectIndex >= $scope.suggestions.length) ||
+        ($scope.suggestions.length !== 0 && ($scope.selectIndex < 0 || $scope.selectIndex >= $scope.suggestions.length))
+      ) {
+        searchText[selectedUserIndex] = selectedUser;
+        $scope.suggestions = [];
```

Comment:
this is a boolean function, however it mutates two things inside its body; shall these mutations be factored out to the outer scope?